AirCasting/Graph/Graph.swift

```diff
@@ -214,6 +225,48 @@ struct Graph: UIViewRepresentable {
             return chartDataEntry
         }) ?? []
         uiView.updateWithEntries(entries: entries, isAutozoomEnabled: isAutozoomEnabled)
+        
+        let limitLines = getLimitLines()
+        if limitLines != midnightLines {
+            midnightLines = limitLines
+            uiView.updateMidnightLines(with: limitLines)
+        }
+    }
+    
+    func getMidnightsPoints(startingDate: Date) -> [Double] {
+        let now = Date()
+        var midnight = Calendar.current.date(bySettingHour: 0, minute: 0, second: 0, of: now)!
+        let components = Calendar.current.dateComponents([.hour, .minute, .second], from: midnight)
+        let firstMeasurementDate = startingDate
+        var midnightPoints: [Double] = []
+        
+        while (midnight > firstMeasurementDate) {
```

Comment:
I'm quite sure this `while` and associated helper vars can be replaced by some map/filter combo (tho it's too complex to come up with solution on the spot)

Reply:
Info for others: I talked with Paweł and decided to leave this code as it is for now :)